sdk/core/AzureCore/Source/Pipeline/Transport/URLSessionTransport.swift

```diff
@@ -95,7 +91,41 @@ public class URLSessionTransport: HTTPTransportStage {
         let responseContext = pipelineRequest.context
         let logger = pipelineRequest.logger
 
+        if let cancellationToken = pipelineRequest.context?.value(forKey: .cancellationToken) as? CancellationToken {
```

Comment:
Maybe just add a helper property `PipelineRequest.isCanceled` that performs this logic since we do this in a few places.

Reply:
I don't think it would really simplify anything. I would still need to retrieve the cancellation token using this syntax to be able to start the timeout. Best case I could simplify 

```Swift
if let cancellationToken = pipelineRequest.context?.value(forKey: .cancellationToken) as? CancellationToken
```

to

```Swift
if let cancellationToken = pipelineRequest.cancellationToken
```

But we don't do that for anything else we embed in the context, so I'm not sure why we would do that now.